<a href="https://colab.research.google.com/github/addicted-ai/Housing-Price-Prediction/blob/main/Housing-Price-Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
# Supress Warnings
import warnings
warnings.filterwarnings('ignore')

In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [6]:
# For train-test split
from sklearn.model_selection import train_test_split

# For Min-Max Scaler
from sklearn.preprocessing import MinMaxScaler

# for automated feature selection using RFE
from sklearn.feature_selection import RFE
from sklearn.linear_model import LinearRegression

# StatsModel for model Building as it gives better Statistics
import statsmodels.api as sm

# For Model Evaluation
from sklearn.metrics import r2_score

In [15]:
df = pd.read_csv('https://raw.githubusercontent.com/addicted-ai/Housing-Price-Prediction/main/melb_data.csv')
df.head(10)

Suburb             Address  ...             Regionname Propertycount
0  Abbotsford        85 Turner St  ...  Northern Metropolitan        4019.0
1  Abbotsford     25 Bloomburg St  ...  Northern Metropolitan        4019.0
2  Abbotsford        5 Charles St  ...  Northern Metropolitan        4019.0
3  Abbotsford    40 Federation La  ...  Northern Metropolitan        4019.0
4  Abbotsford         55a Park St  ...  Northern Metropolitan        4019.0
5  Abbotsford      129 Charles St  ...  Northern Metropolitan        4019.0
6  Abbotsford        124 Yarra St  ...  Northern Metropolitan        4019.0
7  Abbotsford       98 Charles St  ...  Northern Metropolitan        4019.0
8  Abbotsford  6/241 Nicholson St  ...  Northern Metropolitan        4019.0
9  Abbotsford       10 Valiant St  ...  Northern Metropolitan        4019.0

[10 rows x 21 columns]

In [16]:
df.describe(percentiles=[0.1,0.25,0.5,0.75,0.9,0.95,0.97,0.99,1])

Rooms         Price  ...    Longtitude  Propertycount
count  13580.000000  1.358000e+04  ...  13580.000000   13580.000000
mean       2.937997  1.075684e+06  ...    144.995216    7454.417378
std        0.955748  6.393107e+05  ...      0.103916    4378.581772
min        1.000000  8.500000e+04  ...    144.431810     249.000000
10%        2.000000  4.800000e+05  ...    144.875128    2671.000000
25%        2.000000  6.500000e+05  ...    144.929600    4380.000000
50%        3.000000  9.030000e+05  ...    145.000100    6555.000000
75%        3.000000  1.330000e+06  ...    145.058305   10331.000000
90%        4.000000  1.850000e+06  ...    145.108900   14042.000000
95%        5.000000  2.290050e+06  ...    145.153631   14949.000000
97%        5.000000  2.600000e+06  ...    145.190470   17276.330000
99%        5.000000  3.338150e+06  ...    145.269365   21650.000000
100%      10.000000  9.000000e+06  ...    145.526350   21650.000000
max       10.000000  9.000000e+06  ...    145.526350   21650.000000

[14 rows x 13 columns]

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13580 entries, 0 to 13579
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         13580 non-null  object 
 1   Address        13580 non-null  object 
 2   Rooms          13580 non-null  int64  
 3   Type           13580 non-null  object 
 4   Price          13580 non-null  float64
 5   Method         13580 non-null  object 
 6   SellerG        13580 non-null  object 
 7   Date           13580 non-null  object 
 8   Distance       13580 non-null  float64
 9   Postcode       13580 non-null  float64
 10  Bedroom2       13580 non-null  float64
 11  Bathroom       13580 non-null  float64
 12  Car            13518 non-null  float64
 13  Landsize       13580 non-null  float64
 14  BuildingArea   7130 non-null   float64
 15  YearBuilt      8205 non-null   float64
 16  CouncilArea    12211 non-null  object 
 17  Lattitude      13580 non-null  float64
 18  Longti

In [11]:
df.shape

(1460, 80)

- There are 13580 rows & 20 columns in the data set.
- There seems to be some null values in 4 columns. We'll check them.
- Need to see data dictionary for more insight of columns

### Dictionary
- Rooms: Number of rooms
Price: Price in dollars
- Method:
  - S - property sold
  - SP - property sold prior
  - PI - property passed in
  - PN - sold prior not disclosed
  - SN - sold not disclosed
  - NB - no bid
  - VB - vendor bid
  - W - withdrawn prior to auction
  - SA - sold after auction
  - SS - sold after auction price not disclosed
  - N/A - price or highest bid not available.
- Type:
  - br - bedroom(s)
  - h - house,cottage,villa, semi,terrace
  - u - unit, duplex
  - t - townhouse
  - dev site - development site
  - o res - other residential.
- SellerG: Real Estate Agent
- Date: Date sold
- Distance: Distance from CBD
- Regionname: General Region (West, North West, North, North east …etc)
- Propertycount: Number of properties that exist in the suburb.
- Bedroom2 : Scraped # of Bedrooms (from different source)
- Bathroom: Number of Bathrooms
- Car: Number of carspots
- Landsize: Land Size
- BuildingArea: Building Size
- CouncilArea: Governing council for the area